### PR TITLE
fix(curriculum): Update tests to accept multiple valid solutions for step 27 of fcc authors project

### LINKED ISSUE #53159: Update tests to accept multiple valid solutions for step 27 of fcc authors project
### Describe the Issue

We had a camper on the [forum](https://forum.freecodecamp.org/t/didnt-pass-test/663517/4), not able to pass step 27 of the authors project with this code

```js
 <p class="bio">${bio.length > 50? `${bio.slice(0,50)}...` : bio}</p>
```
It is expecting this answer instead

```js
<p class="bio">${bio.length > 50 ? bio.slice(0, 50) + '...' : bio}</p>
```
the tests should be updated to accept either valid answer

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-fetch-and-promises-by-building-an-fcc-authors-page/step-27

### Your code

```js
 <p class="bio">${bio.length > 50? `${bio.slice(0,50)}...` : bio}</p>
```

### Expected behavior

see explanation above

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by iprime2**
Checklist:
- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #53159

image

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md [hunk 1]
```diff
@@ -34,13 +34,13 @@ assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50/)
 If the `bio` text is greater than `50` characters, you should extract the first 50 characters with `slice()` and replace the rest with `...`. Don't forget that indexes are zero-based.
 
 ```js
-assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:/)
+assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:|<p\s*class="bio">\$\{bio\.length\s*>\s*50\s*\?\s*`\$\{bio\.slice\(0,\s*50\)\}\.\.\.`\s*:\s*bio\}<\/p>/);
```
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]: Thank you for your contribution @iprime2  Can you try this code, this is the simplified version.
```suggestion
assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*(`\$\{bio\.slice\(0, 50\)\}\.\.\.`|bio\.slice\(0, 50\) \+ '...')\s*:\s*bio\}<\/p>/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md [hunk 2]
```diff
@@ -34,13 +34,13 @@ assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50/)
...
 ```
 
 If the `bio` text is less than 50 characters, use the `bio` text directly.
 
 ```js
-assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:\s*bio\s*\}<\/p>/)
+assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:\s*bio\s*\}<\/p>|<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*("|')\$\{bio\.slice\(\s*0\,\s*50\s*\)\}...("|')\2\s*\:\s*bio\s*\}<\/p>|/)
```
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:43]: same for this please 
```suggestion
assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*(`\$\{bio\.slice\(0, 50\)\}\.\.\.`|bio\.slice\(0, 50\) \+ '...')\s*:\s*bio\s*\}<\/p>/);
```

**Review by JoyShaheb — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md (hunk 1 above)
Comment (iprime2) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]: let me test this if it works I will commit your suggestions.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md (hunk 2 above)
Comment (iprime2) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:43]: let me test this if it works I will commit your suggestions.

**Comment by iprime2** [comment]
@JoyShaheb I tried your regex suggestion but it did not work for 
```
<p class="bio">${bio.length > 50? `${bio.slice(0,50)}...` : bio}</p>
```
But thanks for the suggestions.

**Comment by iprime2** [comment]
@lasjorg please review my pr.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md (hunk 1 above)
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]:
```suggestion
assert.match(code, /<p\s*class\s*=\s*("|')bio\1\s*>\$\{\s*bio\.length\s*>\s*50\s*\?\s*(?:bio\.slice\(\s*0\s*\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2|`\$\{\s*bio\.slice\(\s*0\s*,\s*50\s*\)\s*\}\.\.\.`)\s*:/);
```
A bit shorter, using alternatives only for the truthy result.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md (hunk 2 above)
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:43]:
```suggestion
assert.match(code, /<p\s*class\s*=\s*("|')bio\1\s*>\$\{\s*bio\.length\s*>\s*50\s*\?\s*(?:bio\.slice\(\s*0\s*\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2|`\$\{\s*bio\.slice\(\s*0\s*,\s*50\s*\)\s*\}\.\.\.`)\s*:\s*bio\s*\}\s*<\/p>/)
```

**Comment by JoyShaheb** [comment]
@iprime2 1 gap after ```<p class="bio">``` is not getting accepted, try adding a `\s*` something to optionally add a gap.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md [hunk 3]
```diff
@@ -28,22 +28,21 @@ assert.match(code, /<p\s*class=("|')bio\1>/);
...
 ```js
-assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50/)
+assert.match(code, /<p\s*class=("|')bio\1>\s*\$\{\s*bio\.length\s*>\s*50/)
 ```
 
 If the `bio` text is greater than `50` characters, you should extract the first 50 characters with `slice()` and replace the rest with `...`. Don't forget that indexes are zero-based.
 
 ```js
-assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:/)
+assert.match(code, /<p\s*class=("|')bio\1>\s*\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:|<p\s*class="bio">\s*\$\{bio\.length\s*>\s*50\s*\?\s*`\$\{bio\.slice\(0,\s*50\)\}\.\.\.`\s*:\s*bio\}<\/p>/);
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]: Looks like something might got lost in the meantime. Committed changes from last suggestions were reverted?

Comment (iprime2) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]: I dont understand..

Comment (gikf) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]: Changes from https://github.com/freeCodeCamp/freeCodeCamp/pull/53178#pullrequestreview-1838559849 appear to be lost.

Comment (iprime2) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]: I look into it.

**Comment by iprime2** [comment]
@gikf The changes are there can please again look into it.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md (hunk 3 above)
Comment (gikf) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:37]:
```suggestion
assert.match(code, /<p\s*class\s*=\s*("|')bio\1\s*>\s*\$\{\s*bio\.length\s*>\s*50\s*\?\s*(?:bio\.slice\(\s*0\s*\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2|`\$\{\s*bio\.slice\(\s*0\s*,\s*50\s*\)\s*\}\.\.\.`)\s*:/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md [hunk 4]
```diff
@@ -28,22 +28,21 @@ assert.match(code, /<p\s*class=("|')bio\1>/);
...
 ```
 
 If the `bio` text is less than 50 characters, use the `bio` text directly.
 
 ```js
-assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:\s*bio\s*\}<\/p>/)
+assert.match(code, /<p\s*class=("|')bio\1>\s*\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2\s*\:\s*bio\s*\}<\/p>|<p\s*class=("|')bio\1>\s*\$\{\s*bio\.length\s*>\s*50\s*\?\s*("|')\$\{bio\.slice\(\s*0\,\s*50\s*\)\}...("|')\2\s*\:\s*bio\s*\}<\/p>|/);
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md:43]:
```suggestion
assert.match(code, /<p\s*class\s*=\s*("|')bio\1\s*>\s*\$\{\s*bio\.length\s*>\s*50\s*\?\s*(?:bio\.slice\(\s*0\s*\,\s*50\s*\)\s*\+\s*("|')\.\.\.\2|`\$\{\s*bio\.slice\(\s*0\s*,\s*50\s*\)\s*\}\.\.\.`)\s*:\s*bio\s*\}\s*<\/p>/);
```

**Review by gikf — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]

**Review by JoyShaheb — APPROVED** [review]

**Pull request merged**